Jakefile.js

```diff
@@ -375,33 +382,52 @@ file(builtGeneratedDiagnosticMessagesJSON,[generatedDiagnosticMessagesJSON], fun
 desc("Generates a diagnostic file in TypeScript based on an input JSON file");
 task("generate-diagnostics", [diagnosticInfoMapTs]);
 
-// Publish nightly
-var configureNightlyJs = path.join(scriptsDirectory, "configureNightly.js");
-var configureNightlyTs = path.join(scriptsDirectory, "configureNightly.ts");
+// Generate version files
+
+var distributeVersionOutputFolder = path.join(builtLocalDirectory, "distributeVersion");
+var distributeVersionJS = path.join(distributeVersionOutputFolder, "distributeVersion.js");
+var distributeVersionTS = path.join(scriptsDirectory, "distributeVersion.ts");
 var packageJson = "package.json";
-var programTs = path.join(compilerDirectory, "program.ts");
+var builtPackageJSON = path.join(distributeVersionOutputFolder, "package.json");
+
+file(distributeVersionTS);
+compileFile(/*outFile*/ distributeVersionJS,
+        /*sources*/ [distributeVersionTS],
+        /*prereqs*/ [],
+        /*prefixes*/ [],
+        /*useBuiltCompiler*/ false,
+        /*noOutFile*/ true,
+        /*generateDeclarations*/ false,
+        /*outDir*/ distributeVersionOutputFolder,
+        /*preserveConstEnums*/ undefined,
+        /*keepComments*/ false,
+        /*noResolve*/ false,
+        /*stripInternal*/ false);
+
+directory(distributeVersionOutputFolder);
+file(builtPackageJSON, [distributeVersionOutputFolder], function() {
+    if (fs.existsSync(distributeVersionOutputFolder)) {
+        console.log("Copy package.json to the buitl\local");
+        jake.cpR(packageJson, builtPackageJSON);

```

Comment:
so package.json contains the truth value of version. It is copy so that when build public_store and inject public_store, Typescript-vs can read the same file and update the version when building typescript-vs